source/jormungandr/jormungandr/street_network/streetnetwork_backend_manager.py

```diff
@@ -86,14 +91,67 @@ def _create_street_network_backends(self, instance, instance_configuration):
 
             backend = utils.create_object(config)
 
-            self._streetnetwork_backends_by_instance[instance].append(backend)
+            self._streetnetwork_backends_by_instance_legacy[instance].append(backend)
             self.logger.info(
                 '** StreetNetwork {} used for direct_path with mode: {} **'.format(
                     type(backend).__name__, backend.modes
                 )
             )
 
-    def get_street_network(self, instance, mode, request):
+    def _create_backend_from_db(self, sn_backend, instance):
+        config = {"class": sn_backend.klass, "args": sn_backend.args}
+        config['args'].setdefault('service_url', None)
+        config['args'].setdefault('instance', instance)
+
+        return utils.create_object(config)
+
+    def _update_sn_backend(self, sn_backend, instance):
+        self.logger.info('Updating / Adding {} streetnetwork backend'.format(sn_backend.id))
+        try:
+            self._streetnetwork_backends[sn_backend.id] = self._create_backend_from_db(sn_backend, instance)
+            self._streetnetwork_backends_last_update[sn_backend.id] = sn_backend.last_update()
+        except Exception:
+            self.logger.exception('impossible to initialize streetnetwork backend')
+
+    def _update_config(self, instance):
+        """
+        Update list of streetnetwork backends from db
+        """
+        if (
+            self._last_update + datetime.timedelta(seconds=self._update_interval) > datetime.datetime.utcnow()
+            or not self._sn_backends_getter
+        ):
+            return
+
+        self.logger.debug('Updating streetnetwork backends from db')
+        self._last_update = datetime.datetime.utcnow()
+
+        sn_backends = []
+        try:
+            sn_backends = self._sn_backends_getter()
+        except Exception:
+            self.logger.exception('Failure to retrieve streetnetwork backends configuration')
+        if not sn_backends:
+            self.logger.debug('No streetnetwork backends/All streetnetwork backends disabled in db')
+            self._streetnetwork_backends = {}
+            self._streetnetwork_backends_last_update = {}
+
+            return
+
+        for sn_backend in sn_backends:
+            if (
+                sn_backend.id not in self._streetnetwork_backends_last_update
+                or sn_backend.last_update() > self._streetnetwork_backends_last_update[sn_backend.id]
+            ):
+                self._update_sn_backend(sn_backend, instance)
+
+    def get_street_network_db(self, instance, streetnetwork_backend_conf):
+        # Make sure we update the streetnetwork_backends list from the database before returning them
+        self._update_config(instance)
+
+        return self._streetnetwork_backends[streetnetwork_backend_conf]
```

Comment:
This will throw if you have a bad backend configuration setup in db. You might want to protect against looking for a wrong key: 
```python
self._streetnetwork_backends.get(streetnetwork_backend_conf)
```